### ISSUE TITLE
Q: What do the NaN in RBD binding data mean for Omicron_BA2, position 392?

### ISSUE THREAD

**Issue by corneliusroemer** [jbloomlab/SARS-CoV-2-RBD_DMS_Omicron#1]
I'm trying to set up a Nextclade prototype using your RBD data contained at: https://media.githubusercontent.com/media/jbloomlab/SARS-CoV-2-RBD_DMS_Omicron/main/results/final_variant_scores/final_variant_scores.csv

It all looks good, but I just noticed that at one position, there are NaNs, and I'm not sure what I should do with these. Are you aware of the fact that there are NaN? Did the experiment for these fail? Are they maybe not viable? Should I assume the delta_rbd is 0 for these - or impute from BA1?

Here's a screenshot of what that slice looks like:
<img width="960" alt="image" src="https://user-images.githubusercontent.com/25161793/191567137-7411479f-2bb2-4a04-adaf-f377bba1bb93.png">

For other targets, e.g. BA1, there seems to be data for this position.

Thanks!

**Comment by jbloom** [comment]
@corneliusroemer, awesome to hear that you are doing this, and let us know any ways we can help.

I believe site 392 is missing values in the ACE2 affinity deep mutational scanning because during the mutagenesis to make the mutants for the BA.2 RBD, this particular site failed (@tylernstarr @ajgreaney, this is correct, right?). There will probably be a few other sites or mutations also missing values in the data set. These should be gray in these heat maps: https://jbloomlab.github.io/SARS-CoV-2-RBD_DMS_Omicron/RBD-heatmaps/

Basically, because the value is missing, it's hard to say what it should be filled too. I think it could be risky to fill to zero depending on purpose. I'm not sure if the way you will display will allow missing values? The other possibility is that I doubt any mutations are missing across **all** the variant backgrounds. Most mutations don't shift much in effects on ACE2 affinity, so values missing in BA.2 could be filled with the measurements for BA.1 (or even Wuhan-Hu-1) pretty accurately in most cases. If this is done, that should just somehow be noted.

**Comment by tylernstarr** [comment]
I agree with @jbloom 

Yes, site 392 failed in library synthesis in the BA.2 background, so those mutants were missing in the experiments.

As Jesse says, filling in missing values to 0 could be misleading as that's not necessarily the default mutaitonal impact.

I agree with Jesse that if you can't tolerate missing values, you could do a decent job imputing missing values from the closest genotype that we do have the measurement in (e.g. BA.1 and BA.2 could be reasonably imputed w.r.t one another)

**Comment by corneliusroemer** [comment]
Thanks both! Imputing with BA.1 or Wuhan or a mix of the two seems like a reasonable solution.

We are calculating one single score - so we do need a value. The good thing about 392 is that I have never seen a sequence with 392 mutated, like ever. covSpectrum agrees - it's very rare, the few sequences may be artefacts.

<img width="991" alt="image" src="https://user-images.githubusercontent.com/25161793/191605607-0d682d85-ca3c-4306-bc90-7b213039c4e9.png">

As far as I can tell it's only that site that has missing values. I would have noticed as I had to manually replace NaNs when I transformed your data to a format we can import into Nextclade.

So far results look promising. Interestingly, BA.2.75 has the highest affinity of the BA.2 sample sequences in the BA.2 dataset. That would explain why it is able to pick up mutations that cause antibody escape but are detrimental to ACE2 binding, like S:486S.

**Comment by jbloom** [comment]
Great. Are you just adding the ACE2 affinity scores to get predicted affinities? That actually should work (in the absence of epistasis) as they are log Kd values which are in principle supposed to be additive in the absence of epistasis.

And yes, I agree that as far as I've been able to track the literature, the actual experiments do indicate BA.2.75 has highest affinity of recent variants, so that's a good sanity check!

**Comment by tylernstarr** [comment]
Yes, and I also share your sentiment that site 392 (a) is invariant (I recall it being highly conserved across broader SARS-related CoV evolution, too), and (b) I believe the only site that had missing mutaitons in the BA.1 and BA.2 libraries. (Wuhan-Hu-1 and some of the earlier variants we profiled had an older version of library generation where we ended up with more missing mutations, in a more sparse pattern. But our Omicron libraries turned out quite nice and homogenous in terms of mutation rates with the exception of BA.2 x 392)

**Comment by corneliusroemer** [comment]
Yes indeed, I'm just adding up whatever substitutions are present.

These are the highest scorers based on consensus pango sequences for each designated pango lineage (from here: https://github.com/corneliusroemer/pango-sequences)

Could still have bugs but it may be correct 🙃 

<img width="794" alt="image" src="https://user-images.githubusercontent.com/25161793/191614520-a0398d6c-26e7-4e2f-9104-98e2df3e21c6.png">

Doesn't look too unreasonable I think.

**Comment by jbloom** [comment]
That's great! 

Two vague thoughts:

1. Emphasize difference between ACE2 affinity and transmissibility. General public seems to get this confused a lot, and think ACE2 affinity = transmissibility whereas (as you noted) it only does up to a threshold, at which point it probably more just supports future antigenic evolution.

2. I do think it would be interesting to see if you could somehow test if the ACE2 affinity of a tree node potentiates its ability to acquire subsequent antigenic mutations.
